pkg/platform/runtime/runtime.go

```diff
@@ -171,16 +172,50 @@ func (r *Runtime) recordCompletion(err error) {
 	var action string
 	if err != nil {
 		action = anaConsts.ActRuntimeFailure
-		if locale.IsInputError(err) {
-			action = anaConsts.ActRuntimeUserFailure
-		}
 	} else {
 		action = anaConsts.ActRuntimeSuccess
 		r.recordUsage()
 	}
 
+	var projectID string
+	if pj, err := model.FetchProjectByName(r.target.Owner(), r.target.Name()); err == nil {
+		projectID = pj.ProjectID.String()
+	} else {
+		multilog.Error("Unable to fetch project by name: %v", errs.JoinMessage(err))
+	}
+
+	ns := project.Namespaced{
+		Owner:   r.target.Owner(),
+		Project: r.target.Name(),
+	}
+
+	var errorType string
+	switch {
+	case locale.IsInputError(err):
+		errorType = "input"
+	case errs.Matches(err, &setup.ProgressReportError{}): // should come before non-input errors
+		errorType = "progress"
+	case errs.Matches(err, &model.SolverError{}):
+		errorType = "solve"
+	case errs.Matches(err, &setup.BuildError{}):
+		errorType = "build"
+	case errs.Matches(err, &setup.ArtifactSetupErrors{}):
```

Comment:
The type you created is called `ArtifactInstallError`. I think SetupErrors is more appropriate though. `ArtifactInstallError` is assuming too many things are install errors. For example I think a download error would also be an install error by the current implementation.

Reply:
Right, all errors that happen during `fetchAndInstallArtifacts()` are aggregated into a single `ArtifactInstallError` object. You can only figure out we had an `ArtifactDownloadError` or `ArtifactInstallError` by iterating through the list of errors in that original object. Yes, an artifact download error is considered an install error, but I'm tagging it as a download error and identifying it in the subsequent loop here.

Reply:
ok, but my point was that the `ArtifactInstallError` type doesn't seem to be serving any purpose? It's only ever constructed here:

https://github.com/ActiveState/cli/pull/2605/files#diff-c415bd62e9cd64f30d585a034cd17ab1f283c67287d63a1448787a2a81efc043R215

But it's never inspected, so I'm not seeing what it is doing? 

Also note where it is constructed it is too aggressively doing so as we cannot assert at that point whether the failure was in fact an install failure as oppose to a download failure.